contracts/dao/VotingIncentive.sol

```diff
@@ -0,0 +1,324 @@
+   * @param _distributorId id of the distributor
+   * @param _startWeek start week
+   * @param _endWeek end week
+   */
+  function addIncentivesBnb(uint16 _distributorId, uint16 _startWeek, uint16 _endWeek) external payable whenNotPaused {
+    require(assetWhitelist[address(0)], "Bnb not whitelisted");
+    require(msg.value > 0, "Invalid amount");
+    require(_distributorId > 0 && _distributorId <= vault.distributorId(), "Invalid distributorId");
+    require(
+      _startWeek >= (vault.getWeek(block.timestamp) + 1) && _startWeek <= _endWeek,
+      "Only current or future weeks"
+    );
+
+    uint256 numberOfWeeks = uint256(_endWeek) - uint256(_startWeek) + 1;
+    uint256 weeklyAmount = msg.value / numberOfWeeks;
+
+    // Ensure input amount is divisible by number of weeks
+    require(weeklyAmount * numberOfWeeks == msg.value, "Input amount not divisible by number of weeks");
+
+    address _asset = address(0);
+    for (uint16 i = _startWeek; i <= _endWeek; ++i) {
+      weeklyIncentives[_distributorId][i][_asset] += weeklyAmount;
+    }
+
+    emit IncentiveAdded(_distributorId, _startWeek, _endWeek, _asset, msg.value);
+  }
+
+  /**
+   * @dev Add incentives for a distributor for a range of weeks
+   * @param _distributorId id of the distributor
+   * @param _startWeek start week
+   * @param _endWeek end week
+   * @param _asset address of the asset
+   * @param _expectAmount expected amount to be added; actual amount may less than this due to rounding
+   */
+  function addIncentives(
+    uint16 _distributorId,
+    uint16 _startWeek,
+    uint16 _endWeek,
+    address _asset,
+    uint256 _expectAmount
+  ) external whenNotPaused {
+    require(assetWhitelist[_asset], "Asset not whitelisted");
+    require(_expectAmount > 0, "Invalid amount");
+    require(_distributorId > 0 && _distributorId <= vault.distributorId(), "Invalid distributorId");
+    require(
+      _startWeek >= (vault.getWeek(block.timestamp) + 1) && _startWeek <= _endWeek,
+      "Only current or future weeks"
+    );
+
+    uint256 numberOfWeeks = uint256(_endWeek) - uint256(_startWeek) + 1;
+    uint256 weeklyAmount = _expectAmount / numberOfWeeks;
+
+    uint256 actualAmount = 0;
+
+    for (uint16 i = _startWeek; i <= _endWeek; ++i) {
+      weeklyIncentives[_distributorId][i][_asset] += weeklyAmount;
+      actualAmount += weeklyAmount;
+    }
+
+    // Transfer the actual amount from payer to this contract
+    require(actualAmount > 0 && actualAmount <= _expectAmount, "Invalid amount");
+    IERC20(_asset).safeTransferFrom(msg.sender, address(this), actualAmount);
+
+    emit IncentiveAdded(_distributorId, _startWeek, _endWeek, _asset, actualAmount);
+  }
+
+  /**
+   * @dev Claim all incentives for a distributor for a week
+   * @param _distributorId id of the distributor
+   * @param _week week number
+   */
+  function batchClaim(uint16 _distributorId, uint16 _week, address[] memory _assets) external whenNotPaused {
```

Comment:
we can make the batch methods like this
input:
`[{
  week
  distributeId,
  assets: [addres1, address2]
},
...]`

return:
`[{
  week,
  distributedId,
  assets: [{
    address1,
    amount1
  },{
    address2,
    amount2
  }]
},
....]`

Reply:
Updated in 4051215ce3c8d5303d5a4438f78fc679a51afb93

Added a new function `getClaimableAmount` to support the query ABI